### DISCUSSION TITLE
All stars are gone !??!

### DISCUSSION THREAD

**Discussion by Firminator** [PartialVolume/shredos.x86_64#71] [Q&A] [answered]
There were 500+ stars on the old repo which is now archived. The new one has 3. Is there a way to transfer those over?

**Comment by PartialVolume** [comment]
Not as far as I'm aware. It's unfortunate and a consequence of the way we build on top of buildroot and then renaming the repository. The stars move with the repository. If I initialised the repository I lose all the stars.

I did think about forcing the new buildroot on top of the old buildroot but then the git history may get messed up and it doesn't really seem a clean way of doing things.

I'm not planning on updating the buildroot version more than once a year as I find it too time consuming. 

The only way I can think of is maybe archive the old 2020.05 repository by NOT renaming it but simply uploading to a newly created repository. Then delete all the file and folders in the 2020.05 repo so it's empty then delete the git history but do not initialise as that will wipe everything then force upload the of the new code to the 2020.05 , then rename 2020.05. to 2021.08 I guess the only risk is that I somehow manage to lose the stars in the process. I'll have to experiment with some other repos and see if that will work.

**Comment by Firminator** [comment]
Can buildroot and all it's ongoing commits and updates not be pulled in automatically into your fork so you don't have to manually re-create the repo once a year. Not being a regular git user but I was under the impression that this is what git can do. Of course I have no idea what the drawbacks of that are or how complicated this is. One drawback would that some new added feature or change in the original buildroot repo can break ShredOS but the plus of automatically pulling it in and the newly supported hardware, security fixes etc should far outweigh this. Plus ShredOS doesn't have a lot of dependancies, so this minimalist approach is actually working for us :)

**Comment by PartialVolume** [comment, marked as answer]
@Firminator Wow, that took me a while to sort the stars issue out. Glad I only have to do that once a year. I'll need to make a few notes on how I did it.

I finally managed to re-arrange the repos so the stars are retained on [https://github.com/PartialVolume/shredos.x86_64](https://github.com/PartialVolume/shredos.x86_64). Basically !
- renamed to current build (without the stars) as 'shredos.x86_64_2021.08.2_only'
- I renamed .git to .git.orig. on my local copy
- Deleted everything in my local copy.
- Copied the latest build minus any .git directory.
- Changed the remotes to point to the correct repo.
- Renamed .git.orig to .git. Did a git commit -a -m "Merge 2021.08.2 onto 2021.05". 
- pushed the repository to Github
- Renamed the shredos.x86_64_2020.05 repo to shredos.x86_64
BINGO - 200 stars and the latest buildroot code.

There are a couple of minor issues. I can't transfer this discussion. So any that can't be moved will get deleted as [this](https://github.com/PartialVolume/shredos.x86_64_2021.08.2_only) repo will be deleted .

**Reply by Firminator** [reply]
This is huge. I'm glad you found a way as it would be a shame to have the stars lost as they represent how much the community is appreciating this project.

**Comment by PartialVolume** [comment]
@Firminator The stars and issues are back and I managed to move all the discussions and issues back too. :+1:

**Reply by Firminator** [reply]
Way to go. This is it. :rocket:
